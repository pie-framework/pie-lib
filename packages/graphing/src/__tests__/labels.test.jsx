import { shallow } from 'enzyme';
import React from 'react';

import Labels, { getTransform } from '../labels';

describe('Labels', () => {
  let w;
  let onChange = jest.fn();
  const wrapper = (extras) => {
    const defaults = {
      classes: {},
      className: 'className',
      onChange,
    };
    const props = { ...defaults, ...extras };
    return shallow(<Labels {...props} />);
  };
  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });
});

describe('getTransform', () => {
  const assertTransform = (side, expected) => {
    it(`returns ${expected} for ${side}`, () => {
      const r = getTransform('left', 100, 100);
      expect(r).toEqual('translate(-20, 50), rotate(-90)');
    });
  };

  assertTransform('left', 'translate(-20, 50), rotate(-90)');
  assertTransform('right', 'translate(130, 50), rotate(90)');
  assertTransform('top', 'translate(50, -20), rotate(0)');
  assertTransform('bottom', 'translate(50, 130), rotate(0)');
});
