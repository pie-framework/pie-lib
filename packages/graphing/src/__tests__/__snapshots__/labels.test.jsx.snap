// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Labels snapshot renders 1`] = `
<Fragment>
  <WithStyles(RawLabel)
    key="left"
    onChange={[Function]}
    side="left"
  />
  <WithStyles(RawLabel)
    key="top"
    onChange={[Function]}
    side="top"
  />
  <WithStyles(RawLabel)
    key="bottom"
    onChange={[Function]}
    side="bottom"
  />
  <WithStyles(RawLabel)
    key="right"
    onChange={[Function]}
    side="right"
  />
</Fragment>
`;
